metricbeat/module/system/process/helper.go

```diff
@@ -401,7 +401,8 @@ func (procStats *ProcStats) GetProcStats() ([]common.MapStr, error) {
 
 func (procStats *ProcStats) includeTopProcesses(processes []Process) []Process {
 	if !procStats.IncludeTop.Enabled ||
-		(procStats.IncludeTop.ByCPU == 0 && procStats.IncludeTop.ByMemory == 0) {
+		(procStats.IncludeTop.ByCPU == 0 && procStats.IncludeTop.ByMemory == 0) ||
+		len(processes) == 1 {
```

Comment:
Isn't this a general problem that will cause a panic anytime that `len(processes) < procStats.IncludeTop.ByCPU` or `len(processes) < procStats.IncludeTop.ByMemory`? ((Try adding a test with two processes.)

I think logic like this for ByCPU and ByMemory would fix it. 

```
		numProcs := procStats.IncludeTop.ByCPU
		if len(processes) < procStats.IncludeTop.ByCPU {
			numProcs = len(processes)
		}
		result = append(result, processes[:numProcs]...)
```

Reply:
👍  Added a more generic fix and more tests to cover this. Remove the special 1 tests as not needed.